src/Domain/Codec/BinaryCodec.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Erdcsharp.Domain.Exceptions;
+using Erdcsharp.Domain.Values;
+
+namespace Erdcsharp.Domain.Codec
+{
+    public class BinaryCodec
+    {
+        private readonly List<IBinaryCodec> _codecs;
+
+        public BinaryCodec()
+        {
+            _codecs = new List<IBinaryCodec>
+            {
+                new NumericBinaryCodec(),
+                new AddressBinaryCodec(),
+                new BooleanBinaryCodec(),
+                new BytesBinaryCodec(),
+                new TokenIdentifierCodec(),
+                new StructBinaryCodec(this),
+                new OptionBinaryCodec(this),
+            };
+        }
+
+        public (IBinaryType Value, int BytesLength) DecodeNested(byte[] data, TypeValue type)
+        {
+            CheckBufferLength(data);
+
+            var codec = _codecs.SingleOrDefault(c => c.Type == type.BinaryType);
+            if (codec == null)
+                throw new BinaryCodecException($"No codec found for {type.BinaryType}");
+
+            var decode = codec.DecodeNested(data, type);
+            return decode;
+        }
+
+        public IBinaryType DecodeTopLevel(byte[] data, TypeValue type)
+        {
+            CheckBufferLength(data);
+
+            var codec = _codecs.SingleOrDefault(c => c.Type == type.BinaryType);
```

Comment:
Extract private function `GetSingleCodec(type)` and also throw exception within that function? (a bit too much repeated code).